### PR TITLE
user switching: log in as another user should log you out first

### PR DESCRIPTION
**Pull request by jakob223**
This PR changes two things: 
 - when you click "log in as another user", you get logged out as your current user. You can go back and log into it again by pressing the back button. This makes a lot of routing simpler in the GUI. I also made the same change to "Create another user" which should fix Y2K-780
 - When you log out in the CLI, you don't get auto-logged-back into another user. This is Y2K-402 and there's extensive discussion there.

Issue: Y2K-402
Issue: Y2K-780
Issue: Y2K-793

cc @keybase/y2ksquad @patrickxb

### CODE REVIEW

File: go/engine/logout.go [hunk 1]
```diff
@@ -17,31 +17,47 @@ func (e *LogoutEngine) Prereqs() Prereqs                 { return Prereqs{} }
...
-	mctx.Debug("Switching to another logged in account: %s", acct)
+	loggedInAccounts := e.filterLoggedIn(accounts)
+
+	if len(loggedInAccounts) > 0 {
+		maybePlural := ""
+		if len(loggedInAccounts) > 1 {
+			maybePlural = "s"
+		}
+		accountsList := ""
+		for idx, acct := range loggedInAccounts {
```
Comment (heronhaye) [inline go/engine/logout.go]: :+1: the little things

File: shared/actions/config/index.tsx [hunk 2]
```diff
@@ -332,7 +332,7 @@ const startLogoutHandshake = (state: Container.TypedState) =>
 // stuff to trigger this due to a timeout if there's no listeners or something
 function* maybeDoneWithLogoutHandshake(state: Container.TypedState) {
   if (state.config.logoutHandshakeWaiters.size <= 0) {
-    yield RPCTypes.loginLogoutRpcPromise({force: false})
+    yield RPCTypes.loginLogoutRpcPromise({force: false, keepSecrets: false})
```
Comment (heronhaye) [inline shared/actions/config/index.tsx]: when does this get called?

Comment (jakob223) [reply shared/actions/config/index.tsx]: When you click "log out" in the GUI and the GUI determines that you're safe to log out

**Comment by jakob223** [comment]
merging over chat test flake

**Pull request merged**